packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -444,9 +449,24 @@ export class MultiSinkLogger extends TelemetryLogger {
 	constructor(
 		namespace?: string,
 		properties?: ITelemetryLoggerPropertyBags,
-		protected loggers: ITelemetryBaseLogger[] = [],
+		loggers: ITelemetryBaseLogger[] = [],
+		tryInheritProperties?: true,
 	) {
-		super(namespace, properties);
+		let realProperties = properties;
+		if (tryInheritProperties === true) {
+			const merge = (realProperties ??= {});
+			loggers
+				.filter((l): l is this => l instanceof TelemetryLogger)
+				.map((l) => l.properties ?? {})
+				.forEach((cv) => {
+					Object.keys(cv).forEach((k) => {
+						merge[k] = { ...cv[k], ...merge?.[k] };
+					});
+				});
+		}
+
+		super(namespace, realProperties);
+		this.loggers = loggers;
```

Comment:
Is this just style? Or why can't we keep the `protected` field definition in the constructor?

Reply:
can't remember if it was lint or compile error but no members can be initialized before the call to super, including in the params list, so this works around that rule.